tests/unit/sagemaker/workflow/test_steps.py

```diff
@@ -127,22 +139,26 @@ def test_training_step(sagemaker_session):
         "Type": "Training",
         "Arguments": {
             "AlgorithmSpecification": {"TrainingImage": IMAGE_URI, "TrainingInputMode": "File"},
+            "HyperParameters": {
+                "batch-size": training_batch_size_parameter,
```

Comment:
I wouldn't expect to see objects in this assertion. Does it work if asserted on `{"Get": "Parameters.TrainingBatchSize"}`?

Reply:
No that won't work. `pipeline.definition()` won't interpolate parameters like that. `pipeline.to_request()` would return `{"Get": "Parameters.TrainingBatchSize"}`. Here's the code that does that https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/workflow/pipeline.py#L248-L258

We have older unit tests that perform similar assertions https://github.com/aws/sagemaker-python-sdk/blob/master/tests/unit/sagemaker/workflow/test_pipeline.py#L187.

Lmk if you think I should change `pipeline.definition()` logic to return parameter expressions instead of objects.

Reply:
That's fine, I just wanted to make sure we have coverage on the conversion of those Parameter objects to the JSON code, looks like it's in `test_pipeline.py`